lib/dependencies/HarmonyExportDependencyParserPlugin.js

```diff
@@ -69,12 +69,15 @@ module.exports = class HarmonyExportDependencyParserPlugin {
 				parser.state.lastHarmonyImportOrder =
 					(parser.state.lastHarmonyImportOrder || 0) + 1;
 				const clearDep = new ConstDependency(
-					"",
+					parser.isAsiPosition(/** @type {Range} */ (statement.range)[0])
+						? ";"
+						: "",
 					/** @type {Range} */ (statement.range)
 				);
 				clearDep.loc = /** @type {DependencyLocation} */ (statement.loc);
 				clearDep.loc.index = -1;
 				parser.state.module.addPresentationalDependency(clearDep);
+				parser.unsetAsiPosition(/** @type {Range} */ (statement.range)[1]);
```

Comment:
I am afraid it is not fix all cases, also we have `asiSafe: boolean` and should use it instead inserting `;` directly

Reply:
Yes, your concerns are valid. I have looked into the asiSafe logic, but due to its complexity, I’m not entirely sure how to implement it correctly in the current code. Therefore, I referred to the implementation in [HarmonyImportDependencyParserPlugin](https://github.com/webpack/webpack/blob/1012ed854681d8c3d10740143378210966aa9816/lib/dependencies/HarmonyImportDependencyParserPlugin.js#L179) to address the issue.